src/Templating/MutableTemplateRegistry.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Templating;
+
+/**
+ * @author Wojciech Błoszyk <wbloszyk@gmail.com>
+ */
+final class MutableTemplateRegistry extends BaseTemplateRegistry implements MutableTemplateRegistryInterface
+{
+    public function setTemplates(array $templates)
```

Comment:
IMO, if we can't add the corresponding type declarations, we should add `NEXT_MAJOR` comments.

Reply:
Since we're adding type everywhere, and templating is already done (https://github.com/sonata-project/SonataAdminBundle/issues/6404) it's a good idea !